pym/bob/input.py

```diff
@@ -2355,11 +2355,21 @@ def prepare(self, inputEnv, sandboxEnabled, inputStates, inputSandbox=None,
         depDiffSandbox = diffSandbox
         depDiffTools = diffTools.copy()
         thisDeps = {}
+        depsEnv = env.derive()
+        depVars = set()
 
-        for dep in self.__deps:
+        __provideDeps = [ env.substitute(p, "providedDep::"+p)
```

Comment:
This is also unexpected. For example `provideVars` uses the final environment of the package after all dependencies were traversed. I would like to have this for `provideDeps` too unless there is a good reason against it.